src/main/java/org/opentripplanner/middleware/models/BugsnagEventRequest.java

```diff
@@ -29,16 +36,62 @@
     /** URL for downloading the report of the requested event data */
     public String url;
 
+    //// Fields added by otp-middleware for our own tracking purposes. ////
+
+    /** How far back the event data request should go. Populates "event.since" in filter. **/
+    public int daysInPast;
+
+    /** Event request project id. This is not provided with the event request response so must be added separately so
+     * that subsequent calls can be made. */
+    public String projectId;
+
     /** This no-arg constructor exists to make MongoDB happy. */
     public BugsnagEventRequest() {
     }
 
+    public static BugsnagEventRequest createFromRequest(HttpResponseValues response, String projectId, int daysInPast)
+        throws JsonProcessingException
+    {
+        BugsnagEventRequest bugsnagEventRequest = JsonUtils.getPOJOFromHttpBody(response, BugsnagEventRequest.class);
+        if (bugsnagEventRequest != null) {
+            // Add the project id for subsequent calls as it is not provided in the response.
+            bugsnagEventRequest.projectId = projectId;
+            bugsnagEventRequest.daysInPast = daysInPast;
+        }
+        return bugsnagEventRequest;
+    }
+
+    /**
+     * Update this request with a more recent request status from Bugsnag. This handles copying over otp-middleware
+     * specific fields and replacing this instance in MongoDB.
+     */
+    public void update(BugsnagEventRequest updatedRequest) {
+        // Update fields maintained separately from the Bugsnag JSON response.
+        this.status = updatedRequest.status;
+        this.total = updatedRequest.total;
+        this.url = updatedRequest.url;
+        // Replace original request with updated request.
+        Persistence.bugsnagEventRequests.replace(id, this);
+    }
+
+    @Override
+    public boolean delete() {
+        return Persistence.bugsnagEventRequests.removeById(id);
+    }
+
+    @JsonIgnore
+    @BsonIgnore
+    public long getTimeWindowEndInMillis() {
+        long timeWindowMillis = TimeUnit.MILLISECONDS.convert(this.daysInPast, TimeUnit.DAYS);
+        return this.dateCreated.getTime() + timeWindowMillis;
```

Comment:
Shouldn't this just be as follows?

```suggestion
        return this.dateCreated.getTime();
```

It would seem that the end of the applicable time window of a request for events would be whenever the event request was created.